### ISSUE TITLE
[java] [6.3.0] PrematureDeclaration false-positive

### ISSUE THREAD

**Issue by boris-petrov** [pmd/pmd#1067]
This is a regression from 6.2.0. We have a million of those warnings. It happens whenever a variable is used in a lambda which is in a return statement:

```java
String foo = "";
return new ArrayList<String>().stream().anyMatch(bar -> foo.equals(bar));
```

Note that the following doesn't trigger the warning:

```java
String foo = "";
return new ArrayList<String>().stream().anyMatch(foo::equals);
```
